cirq-google/cirq_google/calibration/workflow.py

```diff
@@ -819,13 +832,19 @@ def make_zeta_chi_gamma_compensation_for_moments(
     moments in the final circuit.
 
     Args:
-        circuit_with_calibration: A CircuitWithCalibration (likely returned from run_calibrations)
-            whose mapping argument corresponds to the results in the characterizations argument.
+        circuit: Circuit to compensate or instance of CircuitWithCalibration (likely returned from
+            prepare_characterization_for_moments) whose mapping argument corresponds to the results
+            in the characterizations argument. If circuit is passed the method will attempt to
+            match the circuit against a given characterizations. This step is not computationally
+            efficient and can be avoided by passing the pre-calculated instance of
```

Comment:
what do you mean by not efficient? Am I correct in deducing that it has a quadratic runtime where it scans over all the calibrations for each moment? Is this bad in practice?

Reply:
I have changed the docstring.

In the beginning I was fixed on optimizing this computational cost but after looking at this right now it's probably not relevant. This code will not scale to large circuits and more quibts anyway so it's probably fine to do this calculations many times for now. It would probably be also cleaner in terms of API if we did all the compilation like that from the start: this allows to mix different characterizations (XEB/Floquet together as well) and the structure "CircuitsWithCalibrations" would not be necessary.